packages/in_app_purchase/in_app_purchase_storekit/test/fakes/fake_storekit_platform.dart

```diff
@@ -169,6 +171,25 @@ class FakeStoreKitPlatform {
       case '-[InAppPurchasePlugin addPayment:result:]':
         final String id = call.arguments['productIdentifier'] as String;
         final int quantity = call.arguments['quantity'] as int;
+
+        // in case of testing paymentDiscount
+        if (call.arguments['applicationUsername'] == 'userWithDiscount') {
+          if (call.arguments['paymentDiscount'] != null) {
+            final Map<dynamic, dynamic> discountArgument =
+                call.arguments['paymentDiscount'];
+            discountReceived = <String, dynamic>{};
+            // can't cast directly the argument to Map<String,dynamic>, will receive:
+            // PlatformException(error, type '_InternalLinkedHashMap<Object?, Object?>' is not a subtype of type 'Map<String?, dynamic>' in type cast, null, null)
```

Comment:
This comment sounds like you attempted to cast with `as` instead of `cast<...>`.

Reply:
True. Changed it to a `cast<>` now. Thanks!